internal/serialization/compact_serializer.go

```diff
@@ -95,18 +101,26 @@ func (c CompactStreamSerializer) getOrReadSchema(ctx context.Context, input pubs
 	return schema
 }
 
-func MakeSchemasFromConfig(cfg pubserialization.CompactConfig) map[int64]*Schema {
+func MakeSchemasFromConfig(cfg pubserialization.CompactConfig) (map[int64]*Schema, error) {
 	r := map[int64]*Schema{}
 	for _, ser := range cfg.Serializers() {
-		s := makeSchemaFromSerializer(ser)
+		s, err := makeSchemaFromSerializer(ser)
+		if err != nil {
+			return nil, err
+		}
 		r[s.ID()] = s
 	}
-	return r
+	return r, nil
 }
 
-func makeSchemaFromSerializer(ser pubserialization.CompactSerializer) *Schema {
+func makeSchemaFromSerializer(ser pubserialization.CompactSerializer) (schema *Schema, err error) {
+	defer func() {
+		if rec := recover(); rec != nil {
+			err = makeError(rec)
+		}
+	}()
 	sw := NewSchemaWriter(ser.TypeName())
 	// create the zero value for the type in the serializer and build the schema
 	ser.Write(sw, reflect.Zero(ser.Type()).Interface())
-	return sw.Build()
+	return sw.Build(), nil
```

Comment:
where do we return an error in this function? is the deferred function above recovers from panic and return an error automatically?

Reply:
Yes, the following block at the top of this function converts the panic to an `error`:
```
	defer func() {
		if rec := recover(); rec != nil {
			err = makeError(rec)
		}
	}()
```